node/core/provisioner/src/lib.rs

```diff
@@ -286,23 +292,35 @@ type CoreAvailability = BitVec<bitvec::order::Lsb0, u8>;
 async fn send_inherent_data(
 	relay_parent: Hash,
 	bitfields: &[SignedAvailabilityBitfield],
-	candidates: &[BackedCandidate],
 	return_sender: oneshot::Sender<ProvisionerInherentData>,
 	mut from_job: mpsc::Sender<FromJob>,
 ) -> Result<(), Error> {
 	let availability_cores = request_availability_cores(relay_parent, &mut from_job)
 		.await?
 		.await??;
 
+	let candidates: Vec<BackedCandidate> = {
+		let (tx, rx) = oneshot::channel();
+		from_job
+			.send(FromJob::CandidateBacking(
+				CandidateBackingMessage::GetBackedCandidates(relay_parent, tx),
+			))
+			.await
+			.map_err(|err| Error::CandidateBackingMessageSend(err))?;
+		rx.await?
```

Comment:
@rphmeier Something I don't understand: In in #1903, it says

> Provisioner then can fetch full BackedCandidate for any of those

and describes an extension to `GetBackedCandidates` which allows us to restrict the set of backed candidates that are sent. However, I don't get why we'd need/want that: as far as I can tell, the set of backed candidates that Candidate Backing returns here, requesting all of them, is always a strict subset of the candidates we've taken note of in `self.includable_candidates`. Could we just request all candidates here, and get rid of `ProvisionableData::includable` entirely?

Reply:
That gives the provisioner less choice over which candidates to select.